dhis-2/dhis-services/dhis-service-setting/src/main/java/org/hisp/dhis/setting/SystemSettingManager.java

```diff
@@ -73,11 +78,10 @@
      * @param key the system setting key.
      * @return the setting value.
      */
-    Serializable getSystemSetting( SettingKey key );
-
+    @Transactional( readOnly = true )
     default <T extends Serializable> T getSystemSetting( SettingKey key, Class<T> as )
     {
-        return as.cast( getSystemSetting( key ) );
```

Comment:
Here, could we add a check to compare the "as" class type to the class type of the SettingKey enum? That way we could explicitly catch early cases where the argument class is different from the corresponding SettingKey class.

Reply:
done